Mathlib/Algebra/Lie/Weights/Chain.lean

```diff
@@ -364,4 +365,53 @@ lemma chainTop_isNonZero (α β : Weight R L M) (hα : α.IsNonZero) :
 
 end
 
+section
+
+open LieAlgebra
+open Submodule
+
+variable {K L M : Type*} [Field K] [CharZero K] [LieRing L] [LieAlgebra K L]
+  (H : LieSubalgebra K L) [LieRing.IsNilpotent H]
+  [AddCommGroup M] [Module K M] [LieRingModule L M] [LieModule K L M]
+  [IsTriangularizable K H M] [FiniteDimensional K M]
```

Comment:
I think we might as well exit the `LieModule` namespace since we're about to add two parallel lemmas, in different namespaces which are naturally grouped together by their coefficients being a field.

Also `L` and `M` already exist at this point so you're introducing shadowing duplicates (Lean should really warn you about this). We should therefore only add `K` and the typeclasses which involve it.

I therefore suggest:
```suggestion
end LieModule

section Field

open LieAlgebra LieModule Submodule

variable {K : Type*} [Field K] [CharZero K] [LieAlgebra K L]
  (H : LieSubalgebra K L) [LieRing.IsNilpotent H]
  [Module K M] [LieModule K L M]
  [IsTriangularizable K H M] [FiniteDimensional K M]
```